sdk/clientcore/core/src/main/java/io/clientcore/core/util/auth/CompositeChallengeHandler.java

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package io.clientcore.core.util.auth;
+
+import java.util.List;
+
+/**
+ * Composite challenge handler that iterates through a list of handlers to find a valid authorization header.
+ */
+public class CompositeChallengeHandler extends ChallengeHandler {
```

Comment:
Does this need to be a public type? Should we just force usages of this through `ChallengeHandler.of`?

Reply:
Yeah I think it would be better of ChallengeHandler.of(...) factory method.